sources/packages/api/src/services/disbursement-schedule-service/disbursement-schedule-service.ts

```diff
@@ -305,4 +329,127 @@ export class DisbursementScheduleService extends RecordDataModelService<Disburse
+   * @param disbursementScheduleId
+   * @returns Disbursement and Application details.
+   */
+  async getDisbursementAndApplicationDetails(
+    locationId: number,
+    disbursementScheduleId: number,
+  ): Promise<DisbursementSchedule> {
+    return this.repo
+      .createQueryBuilder("disbursementSchedule")
+      .select([
+        "disbursementSchedule.id",
+        "disbursementSchedule.disbursementDate",
+        "disbursementSchedule.coeStatus",
+        "disbursementSchedule.application",
+        "disbursementSchedule.coeDeniedOtherDesc",
+      ])
+      .innerJoinAndSelect("disbursementSchedule.application", "application")
+      .innerJoinAndSelect("application.location", "location")
+      .innerJoinAndSelect("application.student", "student")
+      .innerJoinAndSelect("student.user", "user")
+      .innerJoinAndSelect("application.offering", "offering")
+      .innerJoinAndSelect("offering.educationProgram", "educationProgram")
+      .leftJoinAndSelect(
+        "disbursementSchedule.coeDeniedReason",
+        "coeDeniedReason",
+      )
+      .where("location.id = :locationId", { locationId })
+      .andWhere("application.applicationStatus IN (:...status)", {
+        status: [ApplicationStatus.enrollment, ApplicationStatus.completed],
+      })
+      .andWhere("disbursementSchedule.id = :disbursementScheduleId", {
+        disbursementScheduleId,
+      })
+      .getOne();
+  }
+
+  /**
+   * Summary of disbursement and application for Approval/Denial of COE.
+   * @param locationId
+   * @param disbursementScheduleId
+   * @returns Disbursement and Application Summary.
+   */
+  async getDisbursementAndApplicationSummary(
+    locationId: number,
+    disbursementScheduleId: number,
+  ): Promise<DisbursementSchedule> {
+    return this.repo
+      .createQueryBuilder("disbursementSchedule")
+      .select([
+        "disbursementSchedule.id",
+        "disbursementSchedule.disbursementDate",
+        "application.id",
+        "application.applicationStatus",
+        "application.assessmentWorkflowId",
+      ])
+      .innerJoin("disbursementSchedule.application", "application")
+      .innerJoin("application.location", "location")
+      .where("location.id = :locationId", { locationId })
+      .andWhere("disbursementSchedule.id = :disbursementScheduleId", {
+        disbursementScheduleId,
+      })
+      .andWhere("application.applicationStatus IN (:...status)", {
+        status: [ApplicationStatus.enrollment, ApplicationStatus.completed],
+      })
+      .getOne();
+  }
+
+  /**
+   * Deny COE for an application.
+   ** Note: If an application has 2 COEs, and if the first COE is Rejected then 2nd COE is implicitly rejected.
+   * @param disbursementScheduleIds
+   * @param userId User who denies the COE.
+   */
+  async updateCOEToDeny(
+    applicationId: number,
+    userId: number,
+    coeDeniedReasonId: number,
+    otherReasonDesc: string,
+  ): Promise<void> {
+    this.repo
```

Comment:
We need await or return the Promise here.

Reply:
Fixed.